src/test/scala/com/fulcrumgenomics/umi/DuplexConsensusCallingIntegrationTest.scala

```diff
@@ -0,0 +1,211 @@
+      Stream.range(start=1, end=5).foreach { i => record(builder, s"group3:r$i:RF", start1=104, start2=5, strand1=Minus, strand2=Plus) }
+    }
+
+
+    // group 4: same as group 3, but different RX (C-C instead of A-C)
+    {
+      Stream.range(start=1, end=6).foreach { i => record(builder, s"group4:r$i:FR", start1=5, start2=104, strand1=Plus, strand2=Minus, umiBases="CTG-GTA") }
+      Stream.range(start=1, end=5).foreach { i => record(builder, s"group4:r$i:RF", start1=104, start2=5, strand1=Minus, strand2=Plus, umiBases="GTA-CTG") }
+    }
+
+    // group 5: five FR read pairs and four RF read pair but wrong bases -> wont call consensus since bases are different
+    {
+      Stream.range(start=1, end=6).foreach { i =>
+        record(builder, s"group5:r$i:FR", start1=6, start2=105, strand1=Plus, strand2=Minus).foreach { rec =>
+          rec.bases = "A" * builder.readLength
+        }
+      }
+      Stream.range(start=1, end=5).foreach { i =>
+        record(builder, s"group5:r$i:RF", start1=105, start2=6, strand1=Minus, strand2=Plus).foreach { rec =>
+          rec.bases = "C" * builder.readLength
+        }
+      }
+    }
+
+
+    (builder.toTempFile(), builder.toIndexedSeq)
+  }
+
+  private case class ConsensusPaths
+  (
+    groupedBam: PathToBam          = makeTempFile("grouped.", ".bam"),
+    consensusBam: PathToBam        = makeTempFile("consensus.", ".bam"),
+    filteredBam: PathToBam         = makeTempFile("filtered.", ".bam"),
+    familySizeHistogram: FilePath  = makeTempFile("grouped.", ".family_size_histogram.txt")
+  )
+
+  private def call(minReadsForCalling: Seq[Int], minReadsForFiltering: Seq[Int]): ConsensusPaths = {
+    val paths = ConsensusPaths()
+
+    // Execute the tools
+    captureItAll { () =>
+      new GroupReadsByUmi(input=inputBam, output=paths.groupedBam, familySizeHistogram=Some(paths.familySizeHistogram), strategy=Strategy.Paired).execute()
+      new CallDuplexConsensusReads(input=paths.groupedBam, output=paths.consensusBam, minReads=minReadsForCalling).execute()
+      new FilterConsensusReads(input=paths.consensusBam, output=paths.filteredBam, minReads=minReadsForFiltering, ref=ref, minBaseQuality=10.toByte).execute()
+    }
+
+    paths
+  }
+
+  /** Extracts the duplex's group name from the record's name. */
+  private def toGroup(rec: SamRecord): String = rec.name.split(':').head
+
+
+  "Duplex Consensus Calling" should "support producing a final BAM with duplex consensuses from one strand" in {
+    // duplex consensuses from groups 0-3, and 6
+    val paths = call(minReadsForCalling=Seq(1,1,0), minReadsForFiltering=Seq(1,1,0))
+
+    // grouped BAM should contain all the reads
+    {
+      val groupedRecs = readBamRecs(paths.groupedBam)
+      groupedRecs.length shouldBe 72
+      groupedRecs.map(toGroup).toSet should contain theSameElementsAs this.inputRecords.map(toGroup).toSet
```

Comment:
Ugh, this was super confusing to me until I realized that `toGroup` is really `toFirstColonSeparatedSegment`, and both GroupReadsByUmi and your test data has read names with colons in them.  Maybe here it should use `.map(_.name)`?